controller/control.go

```diff
@@ -515,6 +515,13 @@ func (c *Controller) hasReplica(address string) bool {
 	return false
 }
 
+func (c *Controller) rmReplicaFromRegisteredReplicas(address string) {
```

Comment:
should we also shutdown the frontend?

Reply:
Not required, since we are already checking for the no of replicas before starting frontend.